src/domain/datasets/services/src/dataset_entry_service.rs

```diff
@@ -0,0 +1,285 @@
+            let task_now = now;
+
+            join_set.spawn(async move {
+                let owner_account_id = get_dataset_owner_id(
+                    &task_account_repository,
+                    &dataset_handle.alias.account_name,
+                )
+                .await?;
+                let dataset_entry = DatasetEntry::new(
+                    dataset_handle.id,
+                    owner_account_id.clone(),
+                    dataset_handle.alias.dataset_name,
+                    task_now,
+                );
+
+                task_dataset_entry_repo
+                    .save_dataset_entry(&dataset_entry)
+                    .await
+                    .int_err()
+            });
+        }
+
+        while let Some(join_result) = join_set.join_next().await {
+            let task_res = join_result.int_err()?;
+
+            task_res?;
+        }
+
+        Ok(())
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+impl MessageConsumer for DatasetEntryService {}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+impl MessageConsumerT<DatasetLifecycleMessage> for DatasetEntryService {
+    #[tracing::instrument(
+        level = "debug",
+        skip_all,
+        name = "DatasetEntryService[DatasetLifecycleMessage]"
+    )]
+    async fn consume_message(
+        &self,
+        _: &Catalog,
+        message: &DatasetLifecycleMessage,
+    ) -> Result<(), InternalError> {
+        tracing::debug!(received_message = ?message, "Received dataset lifecycle message");
+
+        match message {
+            DatasetLifecycleMessage::Created(message) => {
+                self.handle_dataset_lifecycle_created_message(message).await
+            }
+
+            DatasetLifecycleMessage::Deleted(message) => {
+                self.handle_dataset_lifecycle_deleted_message(message).await
+            }
+
+            DatasetLifecycleMessage::Renamed(message) => {
+                self.handle_dataset_lifecycle_renamed_message(message).await
+            }
+
+            DatasetLifecycleMessage::DependenciesUpdated(_) => {
+                // No action required
+                Ok(())
+            }
+        }
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+impl InitOnStartup for DatasetEntryService {
+    #[tracing::instrument(
+        level = "debug",
+        skip_all,
+        name = "DatasetEntryService::run_initialization"
+    )]
+    async fn run_initialization(&self) -> Result<(), InternalError> {
+        if self.maybe_initialization_skipper.is_some() {
+            tracing::debug!("Skip initialization: since a skipper is present");
+
+            return Ok(());
+        }
```

Comment:
Why would we need it to be a component, if we can pass a boolean value to `new`

Reply:
No longer relevant